src/core/obj.js

```diff
@@ -1147,8 +1148,33 @@ var XRef = (function XRefClosure() {
               uncompressed: true,
             };
           }
-          var contentLength = skipUntil(buffer, position, endobjBytes) + 7;
-          var content = buffer.subarray(position, position + contentLength);
+          let contentLength, startPos = position + token.length;
+
+          // Find the next "obj" string, rather than "endobj", to ensure that
+          // we won't skip over a new 'obj' operator in corrupt files where
+          // 'endobj' operators are missing (fixes issue9105_reduced.pdf).
+          while (startPos < buffer.length) {
+            let endPos = startPos + skipUntil(buffer, startPos, objBytes) + 4;
```

Comment:
What does this number 4 stand for? Can we make a constant out of it or add a comment?

Reply:
There's existing magic numbers in this code, e.g. https://github.com/mozilla/pdf.js/blob/master/src/core/obj.js#L1150, which is why I figured it wouldn't hurt if more were added.

I'll have to think about this, since comments or meaningful constant names could end up bloating this code quite a bit :-(

Reply:
> What does this number 4 stand for?

`skipUntil` gives us the starting point of `objBytes`, so we need to get the position after it; compare with https://github.com/mozilla/pdf.js/blob/master/src/core/obj.js#L1150.